### PR TITLE
feat(console,phrases,core): add SSO connector creation guide

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Add SSO connector creation guide (the guide comes right after one new SSO connector is created with the creation modal).
The change includes:
1. [core/schemas] Move the definition of the enum `SsoProviderName` to schemas to be reused by the console since we need to decide which form (OIDC metadata form or SAML metadata form) to show on the guide page.
2. [core/test] Add support of optional `partialValidateConfig` query parameter to API `PATCH /api/sso-connectors/:id/config` since on the guide page, we allow users to save partially valid config.
3. [console] Skeleton of guide page, with guides on the left in markdown format, and forms on the right side.
4. [console] `OidcMetadata` component (including `clientId`, `clientSecret`, `issuer` and `scope` fields)
5. [console] `SamlMetadata` component, including:
 - Attribute mapping component: which is for all SAML-protocol-based SSO connectors, for now, its only SamlSsoConnector, but some others are coming later.
 - Saml configure component, can switch between three different ways: configure metadata URL, configure metadata file content, configure manually

TODO:
1. [console] use file upload component for SAML metadata XML file and signing certificate.

## Testing

Tested locally, integration test will be added once the CRUD process of SSO connector on console is finished.
image
image
image
image

## Checklist

~~- [ ] `.changeset`~~
~~- [ ] unit tests~~
~~- [ ] integration tests~~
- [x] necessary TSDoc comments

### CODE REVIEW

**Review by simeng-li — APPROVED** [review]
core updates ProvideName relocation LGTM. Please wait for console approvals

File: packages/console/src/pages/EnterpriseSso/Guide/SamlMetadataForm/SwitchFormatButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import SwitchArrowIcon from '@/assets/icons/switch-arrow.svg';
+import Tick from '@/assets/icons/tick.svg';
+import ActionMenu from '@/ds-components/ActionMenu';
+import { DropdownItem } from '@/ds-components/Dropdown';
+import DynamicT from '@/ds-components/DynamicT';
+
+import * as styles from './index.module.scss';
+
+type Props = {
+  currentValue: FormFormat;
```
Comment (wangsijie) [inline packages/console/src/pages/EnterpriseSso/Guide/SamlMetadataForm/SwitchFormatButton/index.tsx:14]: nit: we usually use `value` and `onChange`

File: packages/console/src/pages/EnterpriseSso/Guide/SamlMetadataForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  signInEndpoint: undefined,
+  entityId: undefined,
+  x509Certificate: undefined,
+};
+
+type Props = {
+  formFormat: FormFormat;
+};
+
+function GenerateFormByFormat({ formFormat }: Props) {
```
Comment (wangsijie) [inline packages/console/src/pages/EnterpriseSso/Guide/SamlMetadataForm/index.tsx:31]: That does not look like a component name.

Comment (darcyYe) [reply packages/console/src/pages/EnterpriseSso/Guide/SamlMetadataForm/index.tsx]: Updated.

**Review by wangsijie — APPROVED** [review]

**Pull request merged**